### ISSUE TITLE
Is it possible to install optifine on 1.16.4 on mcinabox version-p5?

### ISSUE THREAD

**Issue by ChargedHat42838** [AOF-Dev/MCinaBox#819]
I want to boost FPS to 60+ I have only 7-30 FPS on 6 GB ram device 64 bit architecture.

**Comment by ChargedHat42838** [comment]
![Screenshot_2021-01-11-08-29-01-325_com aof mcinabox](https://user-images.githubusercontent.com/70811047/104143847-23e9ae80-53e7-11eb-8457-ab83fe17e65b.jpg)
It's only around 24 FPS on lowest graphics while allocallocating 1024 mb and 748 mb.

**Comment by matt995** [comment]
Well you cant from my tries i imported my pcs optifine

**Comment by ChargedHat42838** [comment]
Matt can you plis send me your optifine 1.16.4 and forge full files complete set?

**Comment by Adam0530** [comment]
> Matt can you plis send me your optifine 1.16.4 and forge full files complete set?
> Through mega
> NZ or mediafire.

Just install forge from mcinabox... then download optifine and put it on mod folder... Ez

**Comment by ChargedHat42838** [comment]
I can't install forge 1.16.4.

**Comment by thebeast67** [comment]
Yeah we also need fabric cuz sodium is better than optifine
Pls also fix mouse freeze its the only reason I have lagg also fix texture cuz for me i have xray lol

**Comment by ThatGifIsNice** [comment]
> I want to boost FPS to 60+ I have only 7-30 FPS on 6 GB ram device 64 bit architecture.

This is your device i have 90 FPS you need to give the game more ram

**Comment by ThatGifIsNice** [comment]
> ![Screenshot_2021-01-11-08-29-01-325_com aof mcinabox](https://user-images.githubusercontent.com/70811047/104143847-23e9ae80-53e7-11eb-8457-ab83fe17e65b.jpg)
> It's only around 24 FPS on lowest graphics while allocating 1024 mb and 748 mb.

Give it more i have 90 fps with 2 GB alocated

**Comment by matt995** [comment]
You cant install forge on 1.16.4 :c

**Comment by AyHa1810** [comment]
> ![Screenshot_2021-01-11-08-29-01-325_com aof mcinabox](https://user-images.githubusercontent.com/70811047/104143847-23e9ae80-53e7-11eb-8457-ab83fe17e65b.jpg)
> It's only around 24 FPS on lowest graphics while allocating 1024 mb and 748 mb.

Bruh RAM isn't the thing which effects video perfomance. Its the SoC (CPU and GPU) of your phone that effects of your phone.

**Comment by AyHa1810** [comment]
Also if I got the time, I'll upload it here.

**Comment by matt995** [comment]
@AyHa1810 you are right.

**Comment by Adam0530** [comment]
> > ![Screenshot_2021-01-11-08-29-01-325_com aof mcinabox](https://user-images.githubusercontent.com/70811047/104143847-23e9ae80-53e7-11eb-8457-ab83fe17e65b.jpg)
> > It's only around 24 FPS on lowest graphics while allocating 1024 mb and 748 mb.
> 
> Bruh RAM isn't the thing which effects video perfomance. Its the SoC (CPU and GPU) of your phone that effects of your phone.

Correct, but you still need ram to run this app... just allocate an enough ram... for me, I am only allocate 960mb ram(or you can does lower then this if your phone does not have enough ram to allocate)... And it does run 30fps+ on my potato phone(not really bruh, but helio p35 is still bad for gaming for me) with mincraft 1.16.4 without forge and optifine

**Comment by matt995** [comment]
Sodium is not better for phones though from my test 

Low graphics optifine : 50 fps stable

                         Sodium : 20 unstable. I was using pojav. I have both.

**Comment by Adam0530** [comment]
> Sodium is not better for phones though from my test 
> 
> Low graphics optifine : 50 fps stable
> 
>                          Sodium : 20 unstable. I was using pojav. I have both.

For you test it does not, but for me it does really good job I got 30 - 40fps but with optifine I got 50 - 90fps

**Comment by FreddyFeuerstein** [comment]
I allways install optifine in mcinabox like this

• Open Termux

• Install vnc server and java

• Install hello minecraft launcher in termux

• Select /sdcard/MCinaBox/gamedir as the new minecraft directory

• Install optifine

• Launch mcinabox

• Select mcbbs as the download source

• Select and play the game

**Comment by FreddyFeuerstein** [comment]
Btw, on high dpi phones the performace in optifine can be sometimes doubled by selecting the internal shaders and setting the shader resolution to 0.5x.
I used that with 1.12.2.
In vanilla i get 25 fps.
With optifine on medium settings and the resolution trick i get around 45 fps with nearly no visible difference.

**Comment by ChargedHat42838** [comment]
I don't need optifine now I allocated 2gb ram and medium settings and FPS was wow 70+. On vanilla

**Comment by ChargedHat42838** [comment]
The problem was my biome blend settings it mus be on medium.

**Comment by Kenzo95** [comment]
> I allways install optifine in mcinabox like this
> 
> • Open Termux
> 
> • Install vnc server and java
> 
> • Install hello minecraft launcher in termux
> 
> • Select /sdcard/MCinaBox/gamedir as the new minecraft directory
> 
> • Install optifine
> 
> • Launch mcinabox
> 
> • Select mcbbs as the download source
> 
> • Select and play the game

Hello, can you say how you installed hello minecraft launcher on termux? I Don't find any reference to replicate your results, thanks a lot

**Comment by FreddyFeuerstein** [comment]
> > I allways install optifine in mcinabox like this
> > • Open Termux
> > • Install vnc server and java
> > • Install hello minecraft launcher in termux
> > • Select /sdcard/MCinaBox/gamedir as the new minecraft directory
> > • Install optifine
> > • Launch mcinabox
> > • Select mcbbs as the download source
> > • Select and play the game
> 
> Hello, can you say how you installed hello minecraft launcher on termux? I Don't find any reference to replicate your results, thanks a lot

Just get the jar file from their github. then put it in a folder reachable by Termux and do:
`
export DISPLAY=:0
vncserver :0
java -jar /path/to/hmcl.jar
`

**Comment by MatheusTgamerPro** [comment]
> Yeah we also need fabric cuz sodium is better than optifine
> Pls also fix mouse freeze its the only reason I have lagg also fix texture cuz for me i have xray lol

For the X-Ray Bug click the F11 key inside the world and you're done.
